adder/util.go

```diff
@@ -14,8 +14,23 @@ import (
 	rpc "github.com/libp2p/go-libp2p-gorpc"
 )
 
+// BlockPutHelper is a helper struct to store information useful for
+// BlockPut and PutDAG.
+type BlockPutHelper struct {
+	dests     []peer.ID
+	rpcClient *rpc.Client
+}
+
+// NewBlockPutHelper creates a BlockPutHelper given an rpc client and allocation peers.
+func NewBlockPutHelper(rpcClient *rpc.Client, dests []peer.ID) *BlockPutHelper {
+	return &BlockPutHelper{
+		dests:     dests,
+		rpcClient: rpcClient,
+	}
+}
+
 // PutBlock sends a NodeWithMeta to the given destinations.
-func PutBlock(ctx context.Context, rpc *rpc.Client, n *api.NodeWithMeta, dests []peer.ID) error {
+func (bph *BlockPutHelper) PutBlock(ctx context.Context, n *api.NodeWithMeta) error {
```

Comment:
Now, since the `Helper` is actually looking a lot like a https://godoc.org/github.com/ipfs/go-ipld-format#NodeAdder, let's move into that direction and rename  `BlockPutHelper` to `BlockAdder`. 

`PutBlock` -> `Add`.

`PutDAG` -> `AddMany`